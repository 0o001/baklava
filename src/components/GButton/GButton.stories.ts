import { storiesOf } from '@storybook/vue';
import GButton from './GButton.vue';

storiesOf('GButton', module)
  .addParameters({ component: GButton })
  .add('Base', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton size="big">Base Button</GButton>
        <GButton size="medium">Base Button</GButton>
        <GButton size="small">Base Button</GButton>
        <GButton outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('secondary', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton variant="secondary">Base Button</GButton>
        <GButton variant="secondary" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('tertiary', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton variant="tertiary">Base Button</GButton>
        <GButton variant="tertiary" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('light', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton variant="light">Base Button</GButton>
      </div>
    `,
  }))
  .add('passive', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton variant="passive">Base Button</GButton>
      </div>
    `,
  }))
  .add('success', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
      <GButton variant="success">Base Button</GButton>
      <GButton variant="success" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('warning', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
      <GButton variant="warning">Base Button</GButton>
      <GButton variant="warning" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('danger', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
      <GButton variant="danger">Base Button</GButton>
      <GButton variant="danger" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('info', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
      <GButton variant="info">Base Button</GButton>
      <GButton variant="info" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('rightIcon', () => ({
    components: { GButton },
    template: '<GButton variant="primary" rightIcon="info">Base Button</GButton>',
  }))
  .add('leftIcon', () => ({
    components: { GButton },
    template: `
      <div>
        <GButton size="big" leftIcon="info">Left Icon</GButton>
        <GButton size="medium" leftIcon="info">Left Icon</GButton>
        <GButton size="small" leftIcon="info">Left Icon</GButton>
      </div>
    `,
  }))
  .add('with Icon', () => ({
    components: { GButton },
    template: `
      <div>
        <GButton size="big" icon="info"></GButton>
        <GButton size="medium" icon="info"></GButton>
        <GButton size="small" icon="info"></GButton>
      </div>
    `,
  }))
  .add('disabled', () => ({
    components: { GButton },
    template: `
      <GButton disabled>Disabled Button</GButton>
    `,
  }))
  .add('fluid', () => ({
    components: { GButton },
    template: `
      <GButton variant="primary" fluid leftIcon="info">Deneme</GButton>
    `,
  }))
  .add('gradient', () => ({
    components: { GButton },
    template: `
      <GButton variant="gradient-purple" leftIcon="play-circle">TUTORIAL TYPE</GButton>
    `,
  })).add('link', () => ({
    components: { GButton },
    template: `
    <GButton variant="link">Link</GButton>
  `,
  }));
