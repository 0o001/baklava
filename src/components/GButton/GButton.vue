<template>
  <button
    v-bind="$attrs"
    v-on="$listeners"
    :class="classNames"
  >
    <div :class="contentClassname">
      <GIcon
        :class="slotClassName"
        :size="iconSize"
        class="-leftIcon"
        v-if="leftIcon"
        :name="leftIcon"
        :color="iconColor"
      />
      <slot />
      <GIcon
        v-if="icon"
        :size="iconSize"
        :name="icon"
        :color="iconColor"
      />
      <GIcon
        :class="slotClassName"
        :size="iconSize"
        class="-rightIcon"
        v-if="rightIcon"
        :name="rightIcon"
        :color="iconColor"
      />
    </div>
  </button>
</template>

<script lang="ts">

import GIcon from '../GIcon/GIcon.vue';
import $_ from '../../utils/dashUtil.ts';

const VALID_BUTTON_VARIANT = [
  'passive',
  'primary',
  'primary-darker',
  'secondary',
  'secondary-darker',
  'tertiary',
  'tertiary-darker',
  'success',
  'success-darker',
  'info',
  'info-darker',
  'warning',
  'warning-darker',
  'danger',
  'danger-darker',
  'light',
  'light-darker',
  'dark',
  'dark-darker',
  'gradient-purple',
  'link',
];

export default {
  name: 'GButton',
  components: {
    GIcon,
  },
  props: {
    variant: {
      type: String,
      default: 'primary',
      validator: function (value: string) {
        if (!VALID_BUTTON_VARIANT.includes(value)) {
          throw new Error(`Invalid prop. Give of of ${VALID_BUTTON_VARIANT.join(',')}`);
        }

        return true;
      },
    },
    size: {
      type: String,
      default: 'big',
    },
    fluid: {
      type: Boolean,
      default: false,
    },
    leftIcon: {
      type: String,
      default: undefined,
    },
    rightIcon: {
      type: String,
      default: undefined,
    },
    icon: {
      type: String,
      default: undefined,
    },
    iconColor: {
      type: String,
      default: 'white',
    },
    outline: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classNames () {
      return {
        'g-button': true,
        '-fluid': this.fluid,
        '-icon': this.icon || this.leftIcon || this.rightIcon,
        '-onlyIcon': !!this.icon,
        [$_(this.variant)]: true,
        '-outline': this.outline,
      };
    },
    contentClassname () {
      return {
        content: true,
        [$_(this.size)]: true,
      };
    },
    slotClassName () {
      return {
        'g-mr-10': !!this.leftIcon || false,
        'g-ml-10': !!this.rightIcon || false,
      };
    },
    iconSize () {
      return {
        big: 20,
        medium: 16,
        small: 12,
      }[this.size];
    },
  },
};
</script>

<style lang="scss" scoped>
.g-button {
  display: inline-block;
  outline: none;
  border: 1px solid transparent;
  border-radius: var(--radius-md);
  padding: 0;

  &:hover {
    cursor: pointer;
  }

  &:disabled {
    opacity: .65;
    pointer-events: none;
    user-select: none;
  }

  .content {
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: var(--font-family-base);
    font-size: var(--font-size-14);
    font-weight: 500;
    line-height: 1;
  }

  &.-fluid {
    width: 100%;
    justify-content: center;
  }

  .-big {
    padding: 12px 30px;
  }

  .-medium {
    padding: 8px 20px;
  }

  .-small {
    padding: 5px 15px;
    font-size: var(--font-size-12);
  }

  &.-icon {
    .-big {
      padding: 10px 30px;
    }
    .-medium {
      padding: 8px 20px;
    }
  }

  &.-onlyIcon {
    line-height: 0;
    .-big {
      padding: 10px;
    }
    .-medium {
      padding: 8px;
    }
    .-small {
      padding: 6px;
    }
  }

  &.-passive {
    background-color: white;
    border: 1px solid var(--mid-grey-100);
    color: var(--light-grey-100);
    .-big {
      padding: 11px 29px;
    }
    .-middle {
      padding: 7px 19px;
    }
    .-small {
      padding: 4px 14px;
    }
  }

  &.-primary {
    background-color: var(--orange-500);
    color: var(--white);

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--orange-500);
      color: var(--orange-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--orange-500);
      }
    }

    &:hover {
      background-color: var(--orange-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-primary-darker {
    background-color: var(--orange-300);
    color: white;
    &:hover {
      background-color: var(--orange-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-secondary {
    background-color: var(--main-grey-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--main-grey-500);
      color: var(--main-grey-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--main-grey-500);
      }
    }

    &:hover {
      background-color: var(--main-grey-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-secondary-darker {
    background-color: var(--main-grey-300);
    color: white;
    &:hover {
      background-color: var(--main-grey-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-tertiary {
    background-color: var(--light-grey-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--light-grey-500);
      color: var(--light-grey-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--light-grey-500);
      }
    }

    &:hover {
      background-color: var(--light-grey-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-tertiary-darker {
    background-color: var(--light-grey-300);
    color: white;
    &:hover {
      background-color: var(--light-grey-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-success {
    background-color: var(--green-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--green-500);
      color: var(--green-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--green-500);
      }
    }

    &:hover {
      background-color: var(--green-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-success-darker {
    background-color: var(--green-300);
    color: white;
    &:hover {
      background-color: var(--green-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-info {
    background-color: var(--blue-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--blue-500);
      color: var(--blue-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--blue-500);
      }
    }

    &:hover {
      background-color: var(--blue-400);
    }
    &.icon {
      color: black;
    }
  }

  &.-info-darker {
    background-color: var(--blue-300);
    color: white;
    &:hover {
      background-color: var(--blue-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-warning {
    background-color: var(--orange-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--orange-500);
      color: var(--orange-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--orange-500);
      }
    }

    &:hover {
      background-color: var(--orange-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-warning-darker {
    background-color: var(--orange-300);
    color: white;
    &:hover {
      background-color: var(--orange-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-danger {
    background-color: var(--red-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--red-500);
      color: var(--red-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--red-500);
      }
    }

    &:hover {
      background-color: var(--red-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-danger-darker {
    background-color: var(--red-300);
    color: white;
    &:hover {
      background-color: var(--red-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-light {
    background-color: var(--bg-grey-500);
    color: black;
    &:hover {
      background-color: var(--bg-grey-400);
    }
    &.icon {
      color: black;
      path {
        fill: black;
      }
    }
  }

  &.-light-darker {
    background-color: var(--bg-grey-300);
    color: white;
    &:hover {
      background-color: var(--bg-grey-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-dark {
    background-color: var(--main-grey-500);
    color: white;
    &:hover {
      background-color: var(--main-grey-400);
    }
    &.icon {
      color: white;
      path {
        fill: white;
      }
    }
  }

  &.-link {
    background-color: transparent;
    color: var( --main-grey-500);
    text-decoration: underline;

    .content {
      font-weight: 400;
    }

    &:hover {
      text-decoration: underline;
    }
    &.icon {
      color: white;
      path {
        fill: white;
      }
    }
  }

  &.-dark-darker {
    background-color: var(--main-grey-300);
    color: white;
    &:hover {
      background-color: var(--main-grey-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-gradient-purple {
    cursor: pointer;
    line-height: 1;
    display: inline-block;
    border: 1px solid transparent !important;
    padding: 0;
    border-radius: var(--radius-md);
    background-image: linear-gradient(180deg,var(--bg-grey-500),var(--bg-grey-500)),linear-gradient(100deg,var(--purple-500),var(--blue-500));
    background-origin: border-box;
    background-clip: content-box,border-box;
    text-decoration: none !important;
    svg {
      stroke: var(--purple-500);
    }
    .content {
      background-image: linear-gradient(96deg,var(--purple-500),var(--blue-500));
      -webkit-background-clip: text;
      background-clip: text;
      -webkit-text-fill-color: transparent;
    }
    &:hover {
      background-image: linear-gradient(280deg,var(--blue-500),var(--purple-500));
      .content {
        background-image: none;
        color: white;
        -webkit-text-fill-color: white;
      }
      svg {
        stroke: white;
      }
    }
    &.icon {
      color: white;
      path {
        fill: white;
      }
    }
  }
}
</style>
