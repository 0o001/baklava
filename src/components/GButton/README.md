# GButton

## Slots

| Name      | Description |
| --------- | ----------- |
| `default` | &nbsp;      |

## Props

| Name         | Type      | Description | Default     |
| ------------ | --------- | ----------- | ----------- |
| `variant`    | `String`  |             | `"primary"` |
| `size`       | `String`  |             | `"big"`     |
| `fluid`      | `Boolean` |             | `false`     |
| `left-icon`  | `String`  |             | `undefined` |
| `right-icon` | `String`  |             | `undefined` |
| `icon`       | `String`  |             | `undefined` |
| `icon-color` | `String`  |             | `"white"`   |
| `outline`    | `Boolean` |             | `false`     |

## Computed Properties

| Name               | Type     | Description                                                                      |
| ------------------ | -------- | -------------------------------------------------------------------------------- |
| `classNames`       | `object` | **Dependencies:** `fluid`, `icon`, `leftIcon`, `rightIcon`, `variant`, `outline` |
| `contentClassname` | `object` | **Dependencies:** `size`                                                         |
| `slotClassName`    | `object` | **Dependencies:** `leftIcon`, `rightIcon`                                        |
| `iconSize`         | `object` | **Dependencies:** `size`                                                         |

