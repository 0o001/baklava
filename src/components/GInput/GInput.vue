<template>
  <div :class="wrapperClass">
    <slot name="left-addon" />
    <GFieldWrapper
      class="body"
      :is-value="!!value"
      :icon-position="iconPosition"
      :is-outline-label="isOutlineLabel"
      :feedback="feedback"
      :label="label"
      :icon="icon"
      :success="success"
      :error="error"
      :disable="disable"
      :size="size"
      @icon-clicked="$emit('icon-clicked', $event)"
    >
      <!--
      @slot You can put select, button and box in this field
      -->
      <input
        :type="type"
        @input="onInput"
        v-on="getListeners"
        v-bind="getAttrs"
        ref="input"
        :disabled="disable"
        :class="labelClass"
        :value="getValue"
      >
      <!--
      @slot You can put select, button and box in this field
      -->
    </GFieldWrapper>
    <slot name="right-addon" />
  </div>
</template>

<script lang="ts">

import ClickOutside from '../../directives/ClickOutside';
import GFieldWrapper from '../GFieldWrapper/GFieldWrapper.vue';

export default {
  name: 'GInput',
  components: {
    GFieldWrapper,
  },
  directives: {
    ClickOutside,
  },
  props: {
    value: {
      type: [String, Number, Date],
      default: () => null,
    },
    type: {
      default: 'text',
      type: String,
    },
    isOutlineLabel: {
      default: false,
      type: Boolean,
    },
    success: {
      default: false,
      type: Boolean,
    },
    error: {
      default: false,
      type: Boolean,
    },
    disable: {
      default: false,
      type: Boolean,
    },
    label: {
      default: '',
      type: String,
    },
    feedback: {
      default: '',
      type: String,
    },
    icon: {
      default: '',
      type: String,
    },
    iconPosition: {
      default: '',
      type: String,
    },
    size: {
      default: 'big',
      type: String,
    },
  },
  computed: {
    getAttrs () {
      const { value, ...rest } = this.$attrs;
      return rest;
    },
    getListeners () {
      const { input, ...rest } = this.$listeners;
      return rest;
    },
    wrapperClass () {
      return {
        'g-input': true,
        'left-addon': !!this.$slots['left-addon'],
        'right-addon': !!this.$slots['right-addon'],
      };
    },
    labelClass () {
      let classValue;
      if (this.error) return 'g-text-red-500';
      if (this.success) return 'g-text-green-500';
      if (this.disable) return 'g-text-dark-grey-500';
      return classValue;
    },
    getValue () {
      return this.value;
    },
  },
  methods: {
    onInput (e: any) {
      this.$emit('input', e.target.value);
    },
  },
};
</script>

<style lang="scss" scoped>
.g-input {
  &.right-addon {
    display: flex;
    > .body {
      width: 100%;
      ::v-deep .g-field-wrapper {
        .content {
          border-top-right-radius: 0;
          border-bottom-right-radius: 0;
        }
      }
    }
    ::v-deep > .g-input {
      .g-field-wrapper .content {
        border-left: 0;
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
        border-top-right-radius: var(--radius-md);
        border-bottom-right-radius: var(--radius-md);
      }
    }
    ::v-deep > .g-button {
      z-index: 2;
      margin-left: -1px;
      height: fit-content;
      border-top-left-radius: 0;
      border-bottom-left-radius: 0;
    }
    ::v-deep > .g-native-select {
      min-width: 20%;
      .g-field-wrapper .content {
        border-left: 0;
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
        border-top-right-radius: var(--radius-md);
        border-bottom-right-radius: var(--radius-md);
      }
    }
  }
  &.left-addon {
    display: flex;
    > .body {
      width: 100%;
      ::v-deep .g-field-wrapper {
        .content {
          border-top-left-radius: 0;
          border-bottom-left-radius: 0;
        }
      }
    }
    ::v-deep > .g-input {
      .g-field-wrapper .content {
        border-right: 0;
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
        border-top-left-radius: var(--radius-md);
        border-bottom-left-radius: var(--radius-md);
      }
    }
    ::v-deep > .g-button {
      height: fit-content;
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
    }
    ::v-deep > .g-native-select {
      min-width: 20%;
      .g-field-wrapper .content {
        border-right: 0;
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
        border-top-left-radius: var(--radius-md);
        border-bottom-left-radius: var(--radius-md);
      }
    }
  }
}
</style>
