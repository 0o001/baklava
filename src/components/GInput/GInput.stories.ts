import { storiesOf } from '@storybook/vue';
import GInput from './GInput.vue';
import GButton from '../GButton';
import GNativeSelect from '../GNativeSelect';

storiesOf('GInput', module)
  .addParameters({ component: GInput })
  .add('Default', () => ({
    components: { GInput, GButton, GNativeSelect },
    props: {
      options: {
        default: [
          {
            value: 'Value1',
            text: 'Text1',
          },
          {
            value: 'Value2',
            text: 'Text2',
          },
        ],
      },
    },
    data () {
      return {
        value: '',
      };
    },
    template: `
    <div style="margin: 40px 200px 200px">
      <GInput class="g-mb-20" label="Email" v-model="value"/>
      <GInput class="g-mb-20" label="Email" :disable="true"/>
      <GInput class="g-mb-20" label="Email" icon="info" />
      <GInput class="g-mb-20" size="small" label="Email" icon="info" />
      <GInput class="g-mb-20" label="Email" feedback="feedback" />
      <GInput class="g-mb-20" label="Email" feedback="feedback" isOutlineLabel/>
      <GInput class="g-mb-20" label="Email" icon="info" icon-position="icon-left" />
      <GInput class="g-mb-20" label="Email">
        <template #right-addon> <GInput label="Email"  />  </template>
      </GInput>
      <GInput class="g-mb-20" label="Email">
        <template #right-addon> <GButton>Test</GButton>  </template>
      </GInput>
      <GInput class="g-mb-20" label="Email">
        <template #right-addon> <GNativeSelect defaultValue="Value1" :options="options">Test</GNativeSelect>  </template>
      </GInput>
      <GInput class="g-mb-20" label="Email">
        <template #left-addon> <GInput label="Email"  />  </template>
      </GInput>
      <GInput class="g-mb-20" label="Email">
        <template #left-addon> <GButton>Test</GButton>  </template>
      </GInput>
      <GInput class="g-mb-20" label="Email">
        <template #left-addon> <GNativeSelect defaultValue="Value1" :options="options">Test</GNativeSelect>  </template>
      </GInput>

     </div>`,
  }))
  .add('States', () => ({
    components: { GInput },
    data () {
      return {
        text: '',
        trimText: '',
      };
    },
    template: `
    <div style="margin: 40px 200px 200px">
    <GInput label="Email" disable  feedback="feedback" isOutlineLabel/>
    <GInput label="Email" disable icon="info"/>
    <GInput label="Email" error  feedback="feedback" isOutlineLabel/>
    <GInput label="Email" error icon="info"/>
    <GInput label="Email" success  feedback="feedback" isOutlineLabel/>
    <GInput label="Email" v-model="text" success icon="info"/>
    <GInput label="Trim Prop" v-model="trimText" trim style="margin: 20px 0"/>
    </div>`,
  }));
