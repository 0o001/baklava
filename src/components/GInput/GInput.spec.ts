import { mount } from '@vue/test-utils';
import GInput from './';
import GButton from '../GButton';

describe('GInput', () => {
  let wrapper: any;

  it('should match snapshot when show is prop is given as false ', () => {
    wrapper = mount(GInput);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should props with default values ​​when no props are set', () => {
    expect(wrapper.vm.wrapperClass).toEqual({
      'g-input': true,
      'left-addon': false,
      'right-addon': false,
    });
  });

  it('should render button given on the left-addon', async () => {
    // when
    wrapper = mount(GInput, {
      slots: {
        'left-addon': `<GButton variant="gradient-purple" leftIcon="play-circle">left</GButton>`,
      },
      stubs: {
        GButton: GButton,
      },
    });

    await wrapper.vm.$nextTick();

    // then
    expect(
      wrapper.vm.$slots['left-addon'][0].componentOptions.children[0].text
    ).toEqual('left');
    expect(wrapper.vm.wrapperClass).toEqual({
      'g-input': true,
      'left-addon': true,
      'right-addon': false,
    });
  });

  it('should render button given on the right-addon', async () => {
    // when
    wrapper = mount(GInput, {
      slots: {
        'right-addon': `<GButton variant="gradient-purple" leftIcon="play-circle">Right</GButton>`,
      },
      stubs: {
        GButton: GButton,
      },
    });

    await wrapper.vm.$nextTick();

    // then
    expect(
      wrapper.vm.$slots['right-addon'][0].componentOptions.children[0].text
    ).toEqual('Right');
    expect(wrapper.vm.wrapperClass).toEqual({
      'g-input': true,
      'left-addon': false,
      'right-addon': true,
    });
  });
});
