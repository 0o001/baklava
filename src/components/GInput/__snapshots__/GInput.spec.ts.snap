// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GInput should match snapshot when show is prop is given as false  1`] = `
<div
  class="g-input"
>
   
  <div
    class="body"
  >
    <div
      class="g-field-wrapper -borderline big"
    >
      <!---->
       
      <div
        class="content"
      >
        <input
          type="text"
        />
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
   
</div>
`;
